### DISCUSSION TITLE
Saving last triggered snippet in a Global Variable

### DISCUSSION THREAD

**Discussion by ziontee113** [L3MON4D3/LuaSnip#728] [Q&A] [answered]
Hi, I'm currently using this mapping to trigger the last triggered LuaSnip snippet:
```lua
vim.keymap.set("i", "<A-CR>", function()
    if LAST_TRIGGERED_SNIPPET then
        vim.api.nvim_input("<A-o>")
        vim.schedule(function()
            ls.snip_expand(LAST_TRIGGERED_SNIPPET)
        end)
    end
end, {})
```
and I put this line `LAST_TRIGGERED_SNIPPET = snip` before the `return` of the functions `expand()` and `snip_expand()` in LuaSnip's `init.lua`

Is there a cleaner way to do this (without modifying LuaSnip's source code)? Thank you very much!

**Comment by L3MON4D3** [comment, marked as answer]
Yes! We support dot-repeat (via vim-repeat), but you can do the same programatically by calling `ls.expand_repeat()`, which will expand the last-expanded snippet at your new cursor-position.

**Reply by ziontee113** [reply]
Thank you very much! It's awesome :+1:
